app/controllers/tabbed_controller.rb

```diff
@@ -8,11 +8,9 @@ def show
     @page_title = @item.title
 
     if params[:tab]
-      fail IndexError unless @item.tabs[params[:tab]]
+      render_404 unless @item.tabs[params[:tab]]
     else
       redirect_to('/' + params[:controller] + '/' + @item.tab_path) if @item.tab_path != params[:id]
     end
   end
-
-  rescue_from IndexError, with: :render_404
```

Comment:
@afred: Without this, we'll be missing IndexErrors thrown by CMLess... but as you note, CMLess should really be more precise. So this is fine, and when we have a more precise error, we'll add it.